plugins/amqp_witness_plugin/amqp_witness_plugin.cpp

```diff
@@ -0,0 +1,122 @@
+#include <eosio/amqp_witness_plugin/amqp_witness_plugin.hpp>
+#include <eosio/chain/types.hpp>
+
+#include <eosio/reliable_amqp_publisher/reliable_amqp_publisher.hpp>
+
+namespace eosio {
+static appbase::abstract_plugin& _amqp_witness_plugin = app().register_plugin<amqp_witness_plugin>();
+
+struct amqp_witness_plugin_impl {
+   enum class witness_signature_over {
+      action_mroot,
+      sig_digest
+   };
+
+   signature_provider_plugin::signature_provider_type signature_provider;
+   unsigned staleness_limit;
+   std::string amqp_server;
+   std::string exchange;
+   witness_signature_over witness_signature_type;
+   bool delete_previous = false;
+
+   std::unique_ptr<reliable_amqp_publisher> rqueue;
+
+   struct amqp_witness_msg {
```

Comment:
I really didn't want to do it because the message is so simple.... but it's probably best practice to version any struct sent

Reply:
You could always encode future extensions in the first byte, replacing `is_sig_digest`.